code/modules/events/carp_migration.dm

```diff
@@ -57,7 +62,8 @@
 	while (i <= num_groups)
 		var/group_size = rand(group_size_min, group_size_max)
 		for (var/j = 1, j <= group_size, j++)
-			spawned_carp.Add(new /mob/living/simple_animal/hostile/retaliate/cavern_dweller(spawn_locations[i]))
+			var/mob/living/simple_animal/hostile/retaliate/cavern_dweller/dweller = new(spawn_locations[i])
+			spawned_dweller += "\ref[dweller]"

```

Comment:
If you're not deleting these at the end of the event, you don't really need to keep track of them. No harm in leaving this as is though.